### ISSUE TITLE
idb installation of app - best match target version 0.0 error

### ISSUE THREAD

**Issue by mattadamson** [facebook/idb#804]
Hi

Has anyone seen this error before or know the root cause please?

The best match /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/DeviceSupport/11.0/DeveloperDiskImage.dmg: 11.0 is not suitable for 0.0

I'm using an iphone 6s iOS 15.7 device to launch a test application.  A few other devices do appear to work though which are nearly identical ie the model and iOS version.  

From

https://github.com/facebook/idb/blob/31b6ed0a8c831043601342f0ecbd9ce7b0858594/FBControlCore/Utility/FBDeveloperDiskImage.m

We see this message output

 return [[FBControlCoreError
    describeFormat:@"The best match %@ is not suitable for %ld.%ld", best, targetVersion.majorVersion, targetVersion.minorVersion]
    fail:error];

Because the target version appears to be read as 0.0.  I can't see where / how it's reading this as 0.0

Many thanks

**Comment by james-3141** [comment]
Yes, we are plagued by this!  It is often caused by not having the correct disk image due to not having xCode up to date, but we also see this when we have a large number of physical devices connected.  Did you solve your problem?

**Comment by james-3141** [comment]
I think we may have made progress on this issue, if it is indeed the same one.  In FBDevice.m:

  if (!_name || overwrite) {
    _name = targetInfo.name;
  }
  if (!_osVersion || overwrite) {
    _osVersion = targetInfo.osVersion;
  }
  **_if (\_productVersion || overwrite) {_**
    _productVersion = targetInfo.productVersion;
  }
  if (!_targetType || overwrite) {
    _targetType = targetInfo.targetType;
  }
  if (!_udid || overwrite) {
    _udid = targetInfo.udid;
  }



Here _productVersion always gets set, getting overwritten with "unknown".  A missing '!' ?
Should it be:

  if (**_!_**_productVersion || overwrite) {
    _productVersion = targetInfo.productVersion;
  }

**Comment by james-3141** [comment]
This seems to result in the target version being set to 0.0.0 and then look for the wrong disk image.
